PyART/catalogs/icc.py

```diff
@@ -1,111 +1,209 @@
+        return
+    
+    def load_puncts(self, fname='puncturetracker-pt_loc..asc'):
+        full_name = os.path.join(self.sim_path,fname)
+        if os.path.exists(full_name):
+            X  = np.loadtxt(full_name)
+            t  = X[:, 8]
+            x0 = X[:,22]
+            y0 = X[:,32]
+            x1 = X[:,23]
+            y1 = X[:,33]
+            x  = x0-x1
+            y  = y0-y1
+            r  = np.sqrt(x**2+y**2)
+            th = -np.unwrap(np.angle(x+1j*y))
+            zeros = 0*t
+            pdict = {'t':t,   'r' :r,  'th':th,
+                     'x0':x0, 'y0':y0, 'z0':zeros,
+                     'x1':x1, 'y1':y1, 'z1':zeros}
+        else:
+            print("Warning: no punctures' tracks found!")
+            pdict = None
+        return pdict
+     
+    def load_multipole_txtfile(self, fname_token, raise_error=True, ellmax=None, nu_norm=False):
+        if ellmax is None: ellmax = self.ellmax
+        files = os_ut.find_fnames_with_token(self.sim_path, fname_token) 
+        if len(files)<1:
+            msg = f'No {fname_token}-files found in {self.sim_path}'
+            if raise_error:
+                raise RuntimeError(msg)
             else:
-                return idx_list[0]
-        else: 
-            return idx_list
+                print(f'Warning! {msg}. Returning empty vars')
+                return {}, None, []
+        tmp = ut.safe_loadtxt(files[0])
+        t   = tmp[:,0]
+        n   = len(t)
+        zeros = np.zeros((n,1)) 
+        mydict = {}
+        for l in range(ellmax+1):
+            for m in range(l+1):
+                mydict[(l,m)] = {'real':zeros, 'imag':zeros, 'A':zeros,
+                                 'p':zeros, 'h':zeros}
+        for f in files:
+            l   = ut.extract_value_from_str(f, 'l')
+            m   = ut.extract_value_from_str(f, 'm')
+            X   = ut.safe_loadtxt(f)
+            flm = -(X[:,1]+1j*X[:,2])
+            if nu_norm:
+                flm /= self.metadata['nu']
+            mydict[(l,m)] = wf_ut.get_multipole_dict(flm)
+        
+        return mydict, t, files 
 
-    def meta_from_name(self,name):
-        return self.catalog_meta[self.idx_from_value(name)]
+    def load_psi4(self, tmax_after_peak=200):
+        dict_psi4, t, files = self.load_multipole_txtfile('mp_psi4', raise_error=True)
+        self._t_psi4 = t 
+        self._psi4lm = dict_psi4
+        self.r_extr  = ut.extract_value_from_str(files[0], 'r')
+        
+        try:
+            tmax_psi4,_,_,_ = self.find_max(wave='psi4lm', height=3e-04)
+            DeltaT_end = self.t_psi4[-1]-(tmax_psi4+tmax_after_peak)
+        except Exception as e:
+            print(f'Error while searching max of psi4 time: {e}')
+            DeltaT_end = 0
+        if DeltaT_end>0:
+            self.cut(DeltaT_end, from_the_end=True, cut_psi4lm=True)
+        pass
     
-    def wave_from_name(self,name):
-        return self.waves[self.idx_from_value(name)]
+    def load_hlm_compute_dothlm(self):
```

Comment:
Similarly, this should just be `load_hlm`. The computation of the dothlm can be performed here, or in a separate method.

Edit: actually, `compute_dothlm` could be a method in the parent waveform class, since it should be general for all waveform families